### PR TITLE
Fix input dialog and expanded results box

### PR DESCRIPTION
**Pull request by DJCrapsody**
As this changes were based on @nahojjjen's fixInfoDialogHeight branch, this PR should supersede https://github.com/farkam135/GoIV/pull/294.

Additional things done:
* Rename "Appraisal narrowing" to "Appraisal Refining"
* Remove "(optional)" as refining is closed by default now
* Fix layout of Appraisal Refining
* Add rotating animation to appraisal refining box arrow
* Change rotating animation to happen before dialog shows/hides
* Fix alignment of Evolution spinner using negative margins
* Fix devolution CP text by checking if result CP is negative

Fixes https://github.com/farkam135/GoIV/issues/272 and fixes https://github.com/farkam135/GoIV/issues/293.

### CODE REVIEW

**Comment by Blaisorblade** [comment]
> Fix alignment of Evolution spinner using negative margins

That's a problem for me, the name of Bulbasaur doesn't fit here.

**Comment by DJCrapsody** [comment]
Yeah, still need adjustments. I will test out till I get the perfect number. Possibly -6 or -7. Will use the post-refactored code in your PR.

File: app/src/main/res/layout/expanded_results_box.xml [hunk 1]
```diff
@@ -119,7 +119,10 @@
             android:layout_height="wrap_content"
             android:layout_weight="1"
             android:popupBackground="#FFF"
-            android:gravity="right"/>
+            android:gravity="right"
+            android:layout_marginRight="-8sp"
+            android:layout_marginTop="-8sp"
+            android:layout_marginBottom="-8sp" />
```
Comment (Blaisorblade) [inline app/src/main/res/layout/expanded_results_box.xml]: Think I'm gonna revert this—doesn't work for me, names get cut. We'll have to see about this separately. @DJCrapsody

**Comment by DJCrapsody** [comment]
Can you post a screenshot so I can try out the adjustment?

**Comment by Blaisorblade** [comment]
Here's the screenshot. Merging the rest meanwhile...

image

**Pull request closed without merging**